content/docs/features/dockerfiles.md

```diff
@@ -0,0 +1,249 @@
+security, enables buildpack interoperability, and preserves the ability to rebase - but it comes at a cost. Base image
+authors must anticipate the OS-level dependencies that will be needed at build- and run-time ahead of time, and this
+isn't always possible. This has been a longstanding source of [discussion](https://github.com/buildpacks/rfcs/pull/173)
+within the CNB project: how can we preserve the benefits of buildpacks while enabling more powerful capabilities?
+
+## Buildpacks and Dockerfiles can work together
+
+Buildpacks are often presented as an alternative to Dockerfiles, but we think buildpacks and Dockerfiles can work
+together. Whereas buildpacks are optimized for creating layers that are efficient and logically mapped to the
+dependencies that they provide, Dockerfiles are the most-used and best-understood mechanism for constructing base images
+and installing OS-level dependencies for containers. The CNB Dockerfiles feature allows Dockerfiles to "provide"
+dependencies that buildpacks "require" through a shared [build plan][TODO], by introducing the concept of image
+extensions.
+
+## What are image extensions?
+
+Image extensions are somewhat like buildpacks, although they are also very different. Their purpose is to generate
+Dockerfiles that can be used to extend the builder or run images prior to buildpacks builds. Like buildpacks, extensions
+participate in the `detect` phase - analyzing application source code to determine if they are needed. During `detect`,
+extensions can contribute to the [build plan][TODO] - recording dependencies that they are able to "provide" (though
+unlike buildpacks, they can't "require" anything). If the provided order contains extensions, the output of `detect`
+will be a group of image extensions and a group of buildpacks that together produce a valid build plan. Image extensions
+only generate Dockerfiles - they don't create layers or participate in the `build` phase.
+
+An image extension could be defined with the following directory:
+
+```
+.
+├── extension.toml <- similar to a buildpack buildpack.toml
+├── bin
+│   ├── detect     <- similar to a buildpack ./bin/detect
+│   ├── generate   <- similar to a buildpack ./bin/build
+```
+
+* The `extension.toml` describes the extension, containing information such as its name, ID, and version.
+* `./bin/detect` is invoked during the `detect` phase. It analyzes application source code to determine if the extension
+  is needed and contributes build plan entries.
+* `./bin/generate` is invoked during the `generate` phase (a new lifecycle phase that happens after `detect`). It
+  outputs either or both of `build.Dockerfile` or `run.Dockerfile` for extending the builder or run image,
+  respectively (in the [initial implementation](#phased-approach), only limited `run.Dockerfile`s are allowed).
+
+For more information, see [authoring an image extension][TODO].
+
+## A platform's perspective
```

Comment:
The platform perspective is probably only of interest to platform operators or buildpack developers.  We should raise this with the learning-team so that they can provide a better structure in which to slot this?